### PR TITLE
4612 [Feature] Create component for Conditions for Enhanced Funding

### LINKED ISSUE #4612: [Feature] Create component for Conditions for Enhanced Funding
### Description and related issues
Create the component for Conditions for Enhanced funding. The APD schema already has a place for these values. The shell for the page has already been created. Add the text and fields as specified in the figma. Validation should only be on when adminCheck is on.

```json
conditionsForEnhancedFunding: {
    enhancedFundingQualification: Boolean,
    enhancedFundingJustification: String
}
```

Design ticket #4600
<details><summary>Design closing-notes</summary>
<p>

## [Conditions for Enhanced Funding Closing Notes](https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=5124%3A21120&t=nGJPGYo8G7Bju18z-1)
As of 3.14.2023
(1) CMS determines the system is likely to provide more efficient, economical. and effective adminstation of the State plan.
(2) The system meets the system requirements, standards and conditions, and performance standards in Part 11 of the State Medicaid Manual, as periodically amended.
(3) The system is compatible with the claims processing and information retrieval used in the administration of Medicare for prompt eligibility verification and for processing claims for persons eligible for both programs.
(4) The system supports the data requirements of quality improvement organizations established under Part B of title XI of the Act. 
(5) The State owns any software that is designed, developed, installed or improved with 90 percent FFP.
(6) The Department has a royalty free, non-exclusive. and irrevocable license to reproduce, publish, or otherwise use and authorize others to use, for Federal Government purposes, software, modifications to software, and documentation that is designed, developed, installed or enhanced with 90 percent FFP. 
(7) The costs of the system are determined in accdordance with the 45 CFR 75, subpart E.
(8) The Medicaid agency agrees in writing to use the system for the period of time specified in the advance planning document approved by CMS or for any shorter period of time that CMS determines justifies the Federal funds invested.
(9) The agency agrees in writing that the information in the system will be safeguarded in accordance with subpart F, part 431 of this subchapter.
(10) Use a modular, flexible approach to systems development, including the use of open interfaces and exposed application programming interfaces; the separation of business rules from core programming, available in both human and machine readable formats.
(11) Align to, and advance increasingly, in MITA maturity for business, architecture, and data.
(12) The agency ensures alignment with, and incorporation of, industry standards adopted by the Office of the National Coordinator for Health IT in accordance with 45 CFR part 170, subpart B: The HIPAA privacy, security and transaction standards; accessibility standards established under section 508 of the Rehabilitation Act, or standards that provide greater accessibility for individuals with disabilities, and compliance with Federal civil rights laws; standards adopted by the Secretary under section 1104 of the Affordable Care Act; and standards and protocols adopted by the Secretary under section 1561 of the Affordable Care Act.
(13) Promote sharing, leverage, and reuse of Medicaid technologies and systems within and among States.
(14) Support accurate and timely processing and adjudications/eligibility determinations and effective communications with providers, beneficiaries, and the public.
(15) Produce transaction data, reports, and performance information that would contribute to program evaluation, continuous improvement in business operations, and transparency and accountability.
(16) The system supports seamless coordination and integration with the Marketplace, the Federal Data Services Hub, and allows interoperability with health information exchanges, public health agencies, human services programs, and community organizations providing outreach and enrollment assistance services as applicable.
(17) For E&E systems, the State must have delivered acceptable MAGI-based system functionality, demonstrated by performance testing and results based on critical success factors, with limited mitigations and workarounds.
(18) The State must submit plans that contain strategies for reducing the operational consequences of failure to meet applicable requirements for all major milestones and functionality.
(19) The agency, in writing through the APD, must identify key state personnel by name, type and time commitment assigned to each project.
(20) Systems and modules developed, installed or improved with 90 percent match must include documentation of components and procedures such that the systems could be operated by a variety of contractors or other users.
(21) For software systems and modules developed, installed or improved with 90 percent match, the State must consider strategies to minimize the costs and difficulty of operating the software on alternate hardware or operating systems.
(22) Other conditions for compliance with existing statutory and regulatory requirements, issued through formal guidance procedures, determined by the Secretary to be necessary to update and ensure proper implementation of those existing requirements.

</p>
</details>

### Mockups or link to Figma
No state: https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=8%3A10833&t=iMdBUhoQmUdKGw0J-1
Yes selected state: https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=8%3A10686&t=iMdBUhoQmUdKGw0J-1
Help drawer open state: https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=8%3A10686&t=iMdBUhoQmUdKGw0J-1
Qualification not selected error state: https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=5129%3A21533&t=iMdBUhoQmUdKGw0J-1
Justification missing error state: https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=5129%3A21206&t=iMdBUhoQmUdKGw0J-1

### Behind the scene changes
* create redux actions that create json patches to add to the patch redux store
* update validation to validate Conditions for Enhanced Funding for MMIS pages
* update admin check on the backend to show errors if the fields aren't valid

### Acceptance criteria

- [ ] the Conditions for Enhanced Funding has been updated with the fields from the figma
- [ ] entering data in the fields saves the values to the APD
- [ ] values that have been entered show up in the fields when the page is reloaded
- [ ] if adminCheck is on, display error messages for invalid field
- [ ] if adminCheck is on and there are invalid field, they should also show up in the admin check panel

### Testing criteria

| Given | When | Then | Covered |
| ----- | ---- | ---- | ------- |
|    the Conditions for Enhanced Funding page   |   values are entered in the fields   |   they are saved to the APD in mongo   |         |
| an APD with the Conditions for Enhanced Funding fields filled in | the page opens | the saved values should display | |
| adminCheck is on | the fields are empty | the error messages should appear under the fields and in the adminCheck panel | |
| adminCheck is open | the fields are filled in | there should be no error messages on the page or in the adminCheck panel | |
| adminCheck is off | the fields are valid or invalid | no error messages are displayed | |

### PR DESCRIPTION
**Pull request by thetif**
Resolves #4612 

### Description
Adds the code for the form view of Conditions for Enhanced Funding

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| the Conditions for Enhanced Funding page | the user clicks the "Review Standards and Conditions Regulation" link | the Standards and Conditions Help Drawer should open | jest |
| the Conditions for Enhanced Funding page | the Standards and Conditions Help Drawer is open and the user clicks close in the drawer | the help drawer should close | jest |
| the Conditions for Enhanced Funding page | the APD has no values | the form should show Qualification radio buttons with no values selected | jest |
| the Conditions for Enhanced Funding page | the APD has No Qualifications | the No radio button should be selected | jest |
| the Conditions for Enhanced Funding page | the APD has Yes Qualifications and a Justification | the Yes radio button should be selected and the Justification should display the Justification text | jest |
| the Conditions for Enhanced Funding page with no values | admin check is on |  the error message "Select an Enhanced Funding Qualification" should be displayed | jest |
| the Conditions for Enhanced Funding page with No Qualifications | admin check is on | no error messages should still be displayed | jest |
| the Conditions for Enhanced Funding page with Yes Qualifications | admin check is on | the error message "Provide an Enhanced Funding Justification" should be displayed | jest |

### Steps to manually verify this change

1. Open an MMIS APD
2. Open an Activity
3. Go to Conditions for Enhanced Funding
4. Verify that the page displays like the figma
5. Click the Review Standards and Conditions Regulation link
6. Verify that the Standards and Conditions Help Drawer opens
7. Verify that the text matches the ticket
8. Verify that clicking the Close button on the help drawer closes the drawer
9. Verify that selecting No works
10. Verify that selecting Yes opens the Justification text area
11. Verify that selecting No/Yes is saved by closing the APD and opening it again
12. Enter data in the Justification area
13. Verify that the Justification is saved by closing the APD and opening it again
14. Open the admin panel
15. Verify that no errors are displayed when the fields are filled out
16. Open another Activity
17. Verify that the error message is present when no Qualification has been selected
18. Verify that selecting No causes the error message to disappear
19. Verify that selecting Yes opens the Justification area and displays an error message for not Justification being present
20. Verify that entering data in the Justification area causes the error message to disappear

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [ ] Chromatic link added above
- [x] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when

- [ ] Code has been reviewed by someone other than the original author

### This pull request is ready to review when QA has

- [ ] Verified the functionality related to the change
- [ ] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [ ] Verified all updated pages with the WAVE tool
- [ ] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by tbolt — CHANGES_REQUESTED** [review]
This code looks good and the component works fine. The adminCheck panel is working but it links to `/enhanced-funding` but the nav item is `/conditions`. I don't have a preference between the naming but they'll need to match.

**Review by Sun-Mountain — APPROVED** [review]
🚀

**Comment by thetif** [comment]
@tbolt I switched them all to enchanced-funding, though I found out it would have been easier to just switch the adminCheck to conditions :D

**Review by tbolt — APPROVED** [review]

File: web/src/components/StandardsAndConditionsHelpDrawer.js
Comment (amyd11) [inline web/src/components/StandardsAndConditionsHelpDrawer.js:1]: Glad you made this it's own component since there's so much content in here

**Review by mirano-darren — APPROVED** [review]
Looks good to me, Nice job! 
| Given | When | Then | Covered |
| ----- | ---- | ---- | ------- |
|    the Conditions for Enhanced Funding page   |   values are entered in the fields   |   they are saved to the APD in mongo   |    ✅      |
| an APD with the Conditions for Enhanced Funding fields filled in | the page opens | the saved values should display |✅ |
| adminCheck is on | the fields are empty | the error messages should appear under the fields and in the adminCheck panel |✅ |
| adminCheck is open | the fields are filled in | there should be no error messages on the page or in the adminCheck panel |✅ |
| adminCheck is off | the fields are valid or invalid | no error messages are displayed |✅ |
| the Conditions for Enhanced Funding page | testing hyperlinks | Link goes to the correct page and opens in a new tab |✅ |

**Review by jeromeleecms — COMMENTED** [review]
Is there a way to adjust the width of the content under the "Yes" option? This area is really cramped (and the Figma has it extending all the way to the right border of the layout). 

Also, can we add one last bit of text after this: 

Describe how this activity will or WILL NOT support the Medicaid Standards and Conditions [42 CFR 433.112](https://www.ecfr.gov/current/title-42/chapter-IV/subchapter-C/part-433/subpart-C/section-433.112). **Ensure all Standards and Conditions are addressed.** (bold added for emphasis and does not need to appear in the final copy)

**Review by SGilliamA1M — APPROVED** [review]
Changes look great. I agree with @jeromeleecms regarding making the width size bigger within the expanded container. @thetif I dropped you a note in Chromatic as well. One small typo in the help text. Everything else looks good!

**Comment by thetif** [comment]
@jeromeleecms and @SGilliamA1M how do the changes look? If everything is good, I will merge.

**Review by jeromeleecms — APPROVED** [review]
Looks good!

**Pull request merged**